perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -44,13 +46,52 @@ enum Flags
     Flag_count
 };
 
-static void SetFlags(u16 Result, bool* FlagArray)
-{
-    FlagArray[Flag_SF] = Result & 0x8000;
-    FlagArray[Flag_ZF] = (Result == 0);
-}
+static void SetFlags(const instruction& Instruction, const u16& LeftOperandValue, 
+    const u16& RightOperandValue, const u16& Result, bool* FlagArray)
+{   
+    size_t HighOrderBit = (Instruction.Operands[1].Register.Count == 1) ? 0x80 : 0x8000;
+    bool Parity = true;
+       
+    switch (Instruction.Op)
+    {
+        case Op_add: [[fallthrough]];
+        case Op_sub: 
+        {
+            if (Instruction.Op == Op_add)
+            {
+                FlagArray[Flag_CF] = ((LeftOperandValue & HighOrderBit) || (RightOperandValue & HighOrderBit))
+                    && !(Result & HighOrderBit);
+                FlagArray[Flag_AF] = ((LeftOperandValue & 0x8) || (RightOperandValue & 0x8))
+                    && !(Result & 0x8);
+                FlagArray[Flag_OF] = ((LeftOperandValue & HighOrderBit) == (RightOperandValue & HighOrderBit))
+                    && ((Result & HighOrderBit) != (LeftOperandValue & HighOrderBit));
```

Comment:
```
(Result & HighOrderBit) != (LeftOperandValue & HighOrderBit)
```
could be replaced by
```
(Result ^ LeftOperandValue) & HighOrderBit
```
It's not super important, but I find it only very slightly better (no intermediary converion to bool, and 2 operations instead of 3).

Reply:
Thanks for pointing that out! I need to get more comfortable with XOR.
Added it here and in 2 other places
1f46ad7f442b0b8f36330045ff2877279d1e1639

Reply:
You can simplify still further if you want. (And probably much further, but you have to stop at some point so that's up to you haha)

The next step would be to write
```
(!((LeftOperandValue ^ RightOperandValue) & HighOrderBit)) && ((Result ^ LeftOperandValue) & HighOrderBit)
```
as
```
(~(LeftOperandValue ^ RightOperandValue) & (Result ^ LeftOperandValue)) & HighOrderBit
```
or something like that. (Didn't check correctness, just wrote it quickly to show the idea, if this isn't correct, there is a correct version similar to this where you factor the highorderbit out).

Reply:
Amazing, I'd never thought about the idea of factoring out with bitwise operations but it makes sense.
This was correct, I also factored `HighOrderBit` out of the sub/cmp case.
8c7044616a192d71c8fcf797847a444fa8474fb2

Reply:
basically it doesn't matter when you use the mask to select the bit you want. You can do your operations on all bits, or only the bit you care about, no difference in the end